### DISCUSSION TITLE
Is there a way to perform a next build without writing files to disk?

### DISCUSSION THREAD

**Discussion by Zcehtro** [vercel/next.js#58974] [Help]
### Summary

I want to perform a build without writing files to disk. Something like:

```bash
next build --no-write
```

My main concern is not the files, but rather if the build process throws any errors. But I don't want to write files every time I check for this.

My reason is that sometimes there are rendering errors that will only pop up during runtime, and since Next.js offers SSR, the runtime logic that might err is already executed at build time.

I searched everywhere but couldn't find anything on how to do this. Out of curiosity I asked on ChatGPT and it suggested using `--dry-run` but it didn't work (bad LLM parsing I guess?).

**Comment by kimjansheden** [comment]
Did you find any solution to this? I too find myself in need of running build without writing to any files, just to check for errors that won't otherwise be found until I actually want to build and deploy.
